### ISSUE TITLE
Problem with publishing containers

### ISSUE THREAD

**Issue by SebastianAchilles** [easybuilders/easybuild-containers#21]
After the merge of #17, #18, #19, #20 not all containers were published correctly to ghcr.io:

```
$ docker pull ghcr.io/easybuilders/rockylinux-8.7
Using default tag: latest
latest: Pulling from easybuilders/rockylinux-8.7
manifest unknown
```

However pulling the container with the arch suffix works:
```
$ docker pull ghcr.io/easybuilders/rockylinux-8.7-amd64
Using default tag: latest
latest: Pulling from easybuilders/rockylinux-8.7-amd64
5461c86c2e54: Already exists 
e186033b39e8: Pull complete 
6c6154feafdc: Pull complete 
ef2a0435976d: Pull complete 
b7d47bc22f57: Pull complete 
Digest: sha256:296ac8d16c25512e23a5c5537fa43932e116d2692ecb1a3e561eb850fb078ffc
Status: Downloaded newer image for ghcr.io/easybuilders/rockylinux-8.7-amd64:latest
ghcr.io/easybuilders/rockylinux-8.7-amd64:latest
```

**Comment by ChristianKniep** [comment]
A manifest list was created. Addressable via tag:

```bash
dind >regctl manifest get ghcr.io/easybuilders/rockylinux-8.7:latest
Name:        ghcr.io/easybuilders/rockylinux-8.7:latest
MediaType:   application/vnd.docker.distribution.manifest.list.v2+json
Digest:      sha256:77498b8f233c4a5db64859bb7a542316a4fc0d5ecea31020a298fe8730b44af2

Manifests:

  Name:      ghcr.io/easybuilders/rockylinux-8.7:latest@sha256:296ac8d16c25512e23a5c5537fa43932e116d2692ecb1a3e561eb850fb078ffc
  Digest:    sha256:296ac8d16c25512e23a5c5537fa43932e116d2692ecb1a3e561eb850fb078ffc
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/amd64

  Name:      ghcr.io/easybuilders/rockylinux-8.7:latest@sha256:06ddafd9e9a1f511f657fb89d86b2ac0c2f63d34c3e1f24c01e92fb6cce18713
  Digest:    sha256:06ddafd9e9a1f511f657fb89d86b2ac0c2f63d34c3e1f24c01e92fb6cce18713
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm64
```
.. and via digest:

```bash
dind > regctl manifest get ghcr.io/easybuilders/rockylinux-8.7@sha256:77498b8f233c4a5db64859bb7a542316a4fc0d5ecea31020a298fe8730b44af2
Name:        ghcr.io/easybuilders/rockylinux-8.7@sha256:77498b8f233c4a5db64859bb7a542316a4fc0d5ecea31020a298fe8730b44af2
MediaType:   application/vnd.docker.distribution.manifest.list.v2+json
Digest:      sha256:77498b8f233c4a5db64859bb7a542316a4fc0d5ecea31020a298fe8730b44af2

Manifests:
*snip*
```

But it seems that the manifests where not uploaded, as they are not accessible.

```bash
dind > regctl manifest get ghcr.io/easybuilders/rockylinux-8.7@sha256:296ac8d16c25512e23a5c5537fa43932e116d2692ecb1a3e561eb850fb078ffc
failed to get manifest ghcr.io/easybuilders/rockylinux-8.7@sha256:296ac8d16c25512e23a5c5537fa43932e116d2692ecb1a3e561eb850fb078ffc: request failed: not found [http 404]: {"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}

dind > regctl manifest get ghcr.io/easybuilders/rockylinux-8.7@sha256:06ddafd9e9a1f511f657fb89d86b2ac0c2f63d34c3e1f24c01e92fb6cce18713
failed to get manifest ghcr.io/easybuilders/rockylinux-8.7@sha256:06ddafd9e9a1f511f657fb89d86b2ac0c2f63d34c3e1f24c01e92fb6cce18713: request failed: not found [http 404]: {"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}
```

Seems that the push of the individual images failed. Depending on what tool created the manifest-list, I'd suspect that they might be private so that the manifest-list creation succeeded for the user but is not available for everyone?

**Comment by ChristianKniep** [comment]
A bit more details on what happens when you do this:
```bash
$ docker pull ghcr.io/easybuilders/rockylinux-8.7
Using default tag: latest
latest: Pulling from easybuilders/rockylinux-8.7
manifest unknown
```

1. `docker pull  ghcr.io/easybuilders/rockylinux-8.7` will add `:latest` as tag and issue a `HEAD` request against the registry. The request won't have a body but the header will include the digest.
2. The next request is a `GET` request using the digest which will result in a JSON blob that includes a list of manifests (like in the first code block above).
3. The client iterates through the manifests until it finds a platform object that matches the local `GOOS` and `GOARCH`.
4. the digest of that request will be used to fetch the actual manifest of the container image. This yields `{"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown"}]}` which gives you the error message in the `docker pull` command.

**Comment by verdurin** [comment]
Just encountered this when trying to use the `CentOS 7` image.
Appending the architecture still applies as a workaround.

**Comment by boegel** [comment]
We should get this fixed properly, looks like those container images are also useful to others: https://stackoverflow.com/questions/75999091/has-ghcr-io-easybuilders-centos-7-9latest-disappeared

**Comment by ysagon** [comment]
I wanted to use those images as well and thought my docker or singularity was too old. And then I found this issue:)

**Comment by boegel** [comment]
Problem is fixed now with the changes in #22, thanks for the help @ChristianKniep! 🥳
